nncf/scopes.py

```diff
@@ -130,78 +130,138 @@ def convert_ignored_scope_to_list(ignored_scope: Optional[IgnoredScope]) -> List
     return results
 
 
+def get_matched_ignored_scope_info(
+    ignored_scope: IgnoredScope, nncf_graphs: List[NNCFGraph]
+) -> Tuple[IgnoredScope, Dict[str, Set[str]]]:
+    """
+    Returns matched ignored scope for provided graphs along with all found matches.
+    The resulted ignored scope consist of all matched rules.
+    The found matches consist of a dictionary with a rule name as a key and matched node names as a value.
+
+    :param ignored_scope: Ignored scope instance.
+    :param nncf_graphs: Graphs.
+    :returns: Matched ignored scope along with all matches.
+    """
+    names, patterns, types, subgraphs_numbers = set(), set(), set(), set()
+    matches = {"names": names, "patterns": set(), "types": set(), "subgraphs": set()}
+
+    for graph in nncf_graphs:
+        node_names = set(node.node_name for node in graph.nodes.values())
```

Comment:
It looks like you don't need to create node_names set If ignored_scope does not have names.

Reply:
done